proto-public/pbmesh/v2beta1/gatewayclassconfig.proto

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v2beta1;
+
+message GatewayClassConfig {
+  optional Consul consul = 1;
```

Comment:
Did you make these all optional because we don't really need these in Consul? Maybe I should do the same for the GatewayClass.